### PR TITLE
Stable rxrust

### PR DESCRIPTION
**Pull request by MarcusGrass**
This makes rxRust runnable on rust stable.
It does a few things:

1. Removes test bencher for test benchmarking meaning that we do not benchmark performance in unittests anymore
2. Removes doc include
2. Rewrites declarative macros to "regular" macros, effectively means that we export all shared macros like `observable_proxy_impl!` for example. I couldn't find a nice way to hide them, it might be possible.
3. Stop using drain filter making the drain operation operate in two steps. Causing a race condition where in the worst case the order of the subscribers are jumbled if subscribers are added during the operation. Alternatively it could cause strange behavior if the thread panics while operating on the subscribers causing some amount of subscribers to be removed but not re-added.
4. Remove inner deref by making separate implementations for Mutex, RefCell, and Box, made the code a lot more verbose than it was before. Someone who's better than me att macros might be able to clean up the duplication better.

One more thing of note, rustfmt.toml requires nightly, so to be able to run `cargo fmt` you need to use rust nightly.

Again feedback, comments, etc are much appreciated!

Ps. this fixes https://github.com/rxRust/rxRust/issues/134

### CODE REVIEW

**Comment by MarcusGrass** [comment]
Haha, alright it seems my fix was unstable, reverted it to basically run and do no checks. checkout https://github.com/rxRust/rxRust/pull/145/commits/76cec97a307c841b3ca9dd5b31b0ab41e9271caa specifically schedurer.rs to check how the tests changed when not using bencher.

**Comment by MarcusGrass** [comment]
> That’s great! There are two points that can be improved.
> 
> 1. Should change the CI use stable version to check and test. Config in `main.yml`
> 2. We can use a stable bench test framework to stable the `bench`, not just remove it.
> 
> And this big work should list in changelog. Should we update the readme to tell user ?

Will fix both, I'll use instead, didn't know there was a stable bencher!
```
[dev-dependencies]
float-cmp = "0.8.0"
bencher = "0.1.5"
```

**Comment by MarcusGrass** [comment]
Reinstated benching, pushed it into the bench commit and added some docs and mergify/main.yaml editions

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,10 +3,16 @@
 ## [0.12.0](https://github.com/rxRust/rxRust/compare/v0.11.0...HEAD)
 ### Features
 - **operator**: add `flatten` operator.
+- **tooling**: Make runnable on rust stable by
```
Comment (M-Adoo) [inline CHANGELOG.md:6]: Add below  `Unreleased` section

**Review by M-Adoo — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (M-Adoo) [reply CHANGELOG.md:6]: All the new change should add in `Unreleased` section `0.12.0` is released

Comment (MarcusGrass) [reply CHANGELOG.md]: My bad, fixed!

**Review by M-Adoo — APPROVED** [review]

**Comment by M-Adoo** [comment]
@mergifyio rebase

**Pull request merged**